lib/wings/orm_converter.rb

```diff
@@ -29,7 +29,9 @@ def self.relationship_keys_for(reflections:)
     #
     # @return [Class]
     def self.base_for(klass:)
-      ModelRegistry.reverse_lookup(klass) || Hyrax::Resource
+      ModelRegistry.reverse_lookup(klass) ||
+        klass.try(:valkyrie_class) ||
```

Comment:
Hmm…this isn't a blocker, but I'd almost assume that if the klass responds to :valkyrie_class, we'd go with that.

I haven't used a lot of `try` recently, but I thought `try` was for when I had a possible nil object. So if `klass` is nil, we'll have a false condition and move on to `Hyrax::Resource`. And if `klass` is not nil we're sending the `:valkyrie_class` message regardless of whether `klass` responds to it.

Reply:
for the basic case of `obj.try(:method_name)` `#try` is functionally `respond_to?(:method_name) && public_send(:method_name)`

so we're returning `.valkyrie_class` if `klass` implements it.

i'm not too picky about whether we respect `ModelRegistry` first, or not. probably we could collapse this behavior to just `klass.try(:valkyrie_class) || Hyrax::Resource`, and do something like this in the Wings setup:

```ruby
class ActiveFedora::Base
  self.valkyrie_class
    ModelRegistry.reverse_lookup(self)
  end
end
```

Reply:
I'm okay with the logic being here. And I learned something new about `try` (which once upon a time behaved a bit differently and less friendly)